packages/components/src/navigator/navigator-screen/component.tsx

```diff
@@ -0,0 +1,167 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type { Ref } from 'react';
+// eslint-disable-next-line no-restricted-imports
+import { motion, MotionProps } from 'framer-motion';
+
+/**
+ * WordPress dependencies
+ */
+import { useContext, useEffect, useState } from '@wordpress/element';
+import { useReducedMotion, useFocusOnMount } from '@wordpress/compose';
+import { isRTL } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import {
+	contextConnect,
+	useContextSystem,
+	WordPressComponentProps,
+} from '../../ui/context';
+import { View } from '../../view';
+import { NavigatorContext } from '../context';
+import type { NavigatorScreenProps } from '../types';
+
+const animationEnterDelay = 0;
+const animationEnterDuration = 0.14;
+const animationExitDuration = 0.14;
+const animationExitDelay = 0;
+
+// Props specific to `framer-motion` can't be currently passed to `NavigatorScreen`,
+// as some of them would overlap with HTML props (e.g. `onAnimationStart`, ...)
+type Props = Omit<
+	WordPressComponentProps< NavigatorScreenProps, 'div', false >,
```

Comment:
What does `WordPressComponentProps` do? In other words, why `NavigatorScreenProps` is not enough as Props type for this component?

Reply:
This is the basic type that we've been using for all recently build component's props ([definition](https://github.com/WordPress/gutenberg/blob/92662feeb220336fb28f8b79869cb251a7db4e27/packages/components/src/ui/context/wordpress-component.ts)).

It augments `NavigatorScreenProps` with all of the props that a `div` HTML element may accept. In this case, the `false` means that we're marking `NavigatorScreen` as non-polymorphic.